src/station_gui.cpp

```diff
@@ -107,6 +107,11 @@ void FindStationsAroundSelection()
 		if (IsTileType(tile, MP_STATION) && GetTileOwner(tile) == _local_company) {
 			T *st = T::GetByTile(tile);
 			if (st == nullptr) continue;
+			if (IsTileType(location.tile, MP_STATION) && GetStationIndex(location.tile) == GetStationIndex(tile)) {
```

Comment:
I think this can be simplified to just:
```suggestion
			if (location.tile == tile) {
```

Reply:
Fixed. Works as intended with this change